### PR TITLE
Install kmesh-cni from kmesh daemon

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
The install includes binary copy, kubeconfig generate.

So we donot require copy kubeconfig to each k8s node anymore.

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
There are still many work todo, including:

- [ ] Give kmesh daemon the right role to r/w namespace or pod.
- [ ] binary install
- [ ] cleanup kubeconfig during uninstall

**Comment by hzxuzhonghu** [comment]
@nlgwcy It is ready for review

File: build.sh [hunk 1]
```diff
@@ -49,6 +49,7 @@ function prepare() {
     export EXTRA_CFLAGS="-O0 -g"    
     
     go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.32.0
+    export PATH="$PATH:$(go env GOPATH)/bin"
```
Comment (hzxuzhonghu) [inline build.sh:52]: cc @VanderChen

File: build/docker/clusterrole.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: kmesh
+  labels:
+    app: kmesh
+rules:
+- apiGroups: [""]
+  resources: ["pods","services","namespaces"]
+  verbs: ["get", "update", "patch"]
```
Comment (nlgwcy) [inline build/docker/clusterrole.yaml:10]: miss watch？

File: pkg/cni_plg/chained.go [hunk 3]
```diff
@@ -172,8 +179,23 @@ func deleteCNIConfig(oldconfig []byte) ([]byte, error) {
 }
 
 func chainedKmeshCniPlugin() error {
-	var err error
-	var newCNIConfig []byte
+	// TODO(hzxuzhonghu): install kmesh cni binary, currently it is installed from start_kmesh.sh
```
Comment (nlgwcy) [inline pkg/cni_plg/chained.go:182]: This comment is obsolete.

File: pkg/cni_plg/plugin/plugin.go [hunk 4]
```diff
@@ -80,48 +86,33 @@ func parseSkelArgs(args *skel.CmdArgs) (*cniConf, *k8sArgs, *cniv1.Result, error
...
 	if err != nil {
 		log.Error(err)
 		return enableKmesh, enableSidecar, err
 	}
 
-	labelValue, ok := namespace.Labels["istio.io/dataplane-mode"]
-
-	if ok && labelValue == "Kmesh" {
+	mode := namespace.Labels["istio.io/dataplane-mode"]
+	if mode == "Kmesh" {
```
Comment (nlgwcy) [inline pkg/cni_plg/plugin/plugin.go:100]: pls ignore case

File: build/docker/clusterrole.yaml (hunk 2 above)
Comment (hzxuzhonghu) [reply build/docker/clusterrole.yaml:10]: I think we donot need `watch` now

Comment (hzxuzhonghu) [reply build/docker/clusterrole.yaml:10]: I added patch, because we may need to patch label on the pod.

File: pkg/cni_plg/chained.go (hunk 3 above)
Comment (hzxuzhonghu) [reply pkg/cni_plg/chained.go:182]: Will remove

File: pkg/cni_plg/plugin/plugin.go (hunk 4 above)
Comment (hzxuzhonghu) [reply pkg/cni_plg/plugin/plugin.go:100]: make sense

**Pull request merged**